### PR TITLE
 Refactor LogIn function 

### PR DESCRIPTION
**Pull request by Pirulax**
Reopening as the previous branch was here, instead on my fork. Sorry about that. Need to get used to commit access

Use std::optional for password in CAccountManager as well.
Remove dead code CStaticFnDefs::LogIn

### CODE REVIEW

**Review by StrixG — APPROVED** [review]
Thank you! Feel free to merge.

**Review by ghost — APPROVED** [review]

File: Server/mods/deathmatch/logic/CAccountManager.cpp [hunk 1]
```diff
@@ -530,7 +530,8 @@ bool CAccountManager::LogIn(CClient* pClient, CClient* pEchoClient, const char*
             pEchoClient->SendEcho(SString("login: Account for '%s' is already in use", szAccountName).c_str());
         return false;
     }
-    if (bCheckPassword && (!IsValidPassword(szPassword) || !pAccount->IsPassword(szPassword)))
```
Comment (qaisjp) [inline Server/mods/deathmatch/logic/CAccountManager.cpp:533]: is the `IsValidPassword` not required anymore?

Comment (qaisjp) [reply Server/mods/deathmatch/logic/CAccountManager.cpp:533]: it looks like this function is checked before passwords are used https://github.com/multitheftauto/mtasa-blue/commit/bec0300e5fd456eeed775c773af7d99372de52cf

Comment (Pirulax) [reply Server/mods/deathmatch/logic/CAccountManager.cpp:533]: Have I deleted it?

Comment (qaisjp) [reply Server/mods/deathmatch/logic/CAccountManager.cpp:533]: yes

Comment (Pirulax) [reply Server/mods/deathmatch/logic/CAccountManager.cpp:533]: My bad... Must have overlooked something.

Comment (qaisjp) [reply Server/mods/deathmatch/logic/CAccountManager.cpp:533]: test 123

Comment (Pirulax) [reply Server/mods/deathmatch/logic/CAccountManager.cpp:533]:
```cpp
bool CAccountManager::IsValidPassword(const SString& strPassword)
{
    if (strPassword.length() < MIN_PASSWORD_LENGTH)
        return false;
    return true;
}
```
`IsPassword` will check if the password is empty, so this check was redundant.
Should I add it back anyways?

Comment (qaisjp) [reply Server/mods/deathmatch/logic/CAccountManager.cpp:533]: Probably a good idea

**Pull request merged**